bsb/morphologies/__init__.py

```diff
@@ -1042,6 +1064,52 @@ def attach_child(self, branch):
         self._children.append(branch)
         branch._parent = self
 
+    def find_closest_point(self, coord):
+        """
+        Return the index of the closest on this branch to a desired coordinate.
+
+        :param coord: The coordinate to find the nearest point to
+        :type: :class:`numpy.ndarray`
+        """
+        diff = np.sqrt(np.sum((self._points - coord) ** 2, axis=1))
+        return np.argmin(diff)
+
+    def insert_branch(self, branch, coord):
+        """
+        Insert a new branch on the closest current branch point to a desired coordinate.
+
+        :param branch: Branch to be attached
+        :type branch: :class:`Branch <.morphologies.Branch>`
+        :param point: Coordinates to insert the new branch at or index of the corresponding branch point
+        :type: Union[:class:`numpy.ndarray`, int]
+        """
+        if isinstance(coord, int):
+            index = coord
+        else:
+            index = self.find_closest_point(coord)
```

Comment:
The API is index-centric; so the param name should be `index`, but I don't mind if you want to auto convert coords to indices:

```suggestion
    def insert_branch(self, branch, index):
        """
        Insert a new branch, by splitting this branch and inserting ``branch`` as a child branch at the cutpoint.

        :param branch: Branch to be attached
        :type branch: :class:`Branch <.morphologies.Branch>`
        :param index: Index of the cutpoint; if coordinates are given, the closest point to that coordinate is used
        :type: Union[int, :class:`numpy.ndarray`]
        """
        index = np.array(index, copy=False)
        if index.ndim != 0:
            index = self.find_closest_point(index)
```

Reply:
Might be more consistent to keep it index-centric 👍🏼